ui/src/app/runtime/runtime-apps.service.ts

```diff
@@ -5,55 +5,63 @@ import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
 import {Page} from '../shared/model/page';
 import {RuntimeApp} from './model/runtime-app';
-import {PageInfo} from '../shared/model/pageInfo';
 import {ErrorHandler} from '../shared/model/error-handler';
 
 /**
  * Service to interact with the SCDF server on Runtime applications queries.
  *
  * @author Ilayaperumal Gopinathan
+ * @author Damien Vitrac
  */
 @Injectable()
 export class RuntimeAppsService {
 
-    private runtimeServiceURL = '/runtime/apps';
+  /**
+   * URL API fo runtime Services
+   */
+  private runtimeServiceURL = '/runtime/apps';
 
-    constructor(private http: Http, private errorHandler: ErrorHandler) {
-    }
+  /**
+   * Constructor
+   *
+   * @param {Http} http
+   * @param {ErrorHandler} errorHandler
+   */
+  constructor(private http: Http, private errorHandler: ErrorHandler) {
+  }
 
-    /**
-     * Get the runtime applications based on the paged parameters.
-     *
-     * @param pageInfo the page information for the current page being retrieved.
-     * @returns {Observable<R|T>} the Promise that returns the paged result of Runtime applications.
-     */
-    public getRuntimeApps(pageInfo: PageInfo): Observable<Page<RuntimeApp>> {
-        const params = {};
-        params['page'] = pageInfo.pageNumber.toString();
-        params['size'] = pageInfo.pageSize.toString();
-        return this.http.get(this.runtimeServiceURL, {params: params})
-            .map(this.extractData.bind(this))
-            .catch(this.errorHandler.handleError);
-    }
+  /**
+   * Get the runtime applications based on the paged parameters.
+   *
+   * @returns {Observable<R|T>} the Promise that returns the paged result of Runtime applications.
+   * @param pagination
+   */
+  public getRuntimeApps(pagination: { page: number, size: number }): Observable<Page<RuntimeApp>> {
```

Comment:
I am a little hesitant - Would maybe a `PageInfo` defined as interface work? I think that then you can just pass in a JSON object as parameter as long as it adheres to the *Interface*.

I kind of like having a well-defined interface/class here as I can then introspect the app and ask question like - show me all the methods in the app that perform pagination and pagination itself follows a certain contract.

Reply:
Yes. The interface is in the PR 613 (Pagination, Filter, Sort Interface).